semantics/src/namespace/types.rs

```diff
@@ -219,13 +242,48 @@ impl Array {
     }
 }
 
+impl Tuple {
+    pub fn empty() -> Tuple {
+        Tuple { items: vec![] }
+    }
+
+    pub fn to_fixed_size(&self) -> FixedSize {
+        FixedSize::Tuple(self.clone())
+    }
+}
+
+impl FeSized for Tuple {
+    fn size(&self) -> usize {
+        self.items.iter().map(|typ| typ.size()).sum()
+    }
+}
+
+impl AbiEncoding for Tuple {
+    fn abi_name(&self) -> String {
+        "tuple".to_string()
+    }
+
+    fn abi_size(&self) -> usize {
+        self.size()
+    }
+
+    fn abi_padding(&self) -> AbiPadding {
+        unimplemented!();
+    }
+
+    fn abi_type(&self) -> AbiType {
+        AbiType::Terminal
```

Comment:
Currently, we have `Terminal` and `UniformRecursive` ABI types. I think when we get around to adding support for non-empty tuples, we'll want to create a new ABI type named `Recursive` that contains any set of other ABI types.

For now, I think all of these ABI functions should be marked as unimplemented.